vignettes/EIRprevmatch.Rmd

```diff
@@ -15,68 +15,152 @@ knitr::opts_chunk$set(
+
+In this vignette, three methods for matching the EIR to a target *Pf*PR~2-10~ are outlined and compared. The first uses the `malariaEquilibrium` package's `human_equilibrium()` function to quickly calculate the equilibrial *Pf*PR~2-10~ for a given EIR using the canonical equilibrium solution for the `malariasimulation` model. The second method involves running `malariasimualtion` simulations across a range of initial EIR values, extracting the *Pf*PR~2-10~ from each simulation run, fitting a model relating initial EIR to *Pf*PR~2-10~, and using the model to predict the initial EIR value required to yield the desired *Pf*PR~2-10~. The third approach is to calibrate the model using the `cali` package's (see <https://github.com/mrc-ide/cali> for more information) `calibrate()` function, which searches a user-defined initial EIR parameter space and identifies the value which yields a target *Pf*PR~2-10~ within a defined tolerance.
+
+## *Pf*PR~2-10~ matching using malariaEquilibrium
+
+The `malariaEquilibrium` package function `human_equilibrium()` can be used to return the canonical equilibrium solution of the system for a given EIR and the matching *Pf*PR~2-10~ calculated from the output (see <https://github.com/mrc-ide/malariaEquilibrium> for more information). First, we generate a large vector of EIR values for which we want to generate matching *Pf*PR~2-10~ values. Next, we load the package's default parameter set (`Jamie_parameters.rds`) and, for each EIR value generated, run the `human_equilibrium()` function. The `human_equilibrium()` function returns a dataframe containing the proportion of each age class in each state variables at equilibrium. The *Pf*PR~2-10~ is then calculated from the output for each EIR value by summing the proportion of people aged 2-10 with cases of malaria (`pos_M`) to get the proportion of individuals aged 2-10 with malaria, and dividing this proportion by the proportion of the population between the ages 2-10 (`prop`). Finally, we store the matching EIR and *Pf*PR~2-10~ values in a data frame.
+
+```{r}
+
+# Establish a range of EIR values to generate matching PfPR2-10 values for
+malEq_EIR <- seq(from = 0.1, to = 50, by = 0.5)
+
+# Load the base malariaSimulation parameter set:
+eq_simparams <- malariaEquilibrium::load_parameter_set("Jamie_parameters.rds")
+
+# Use human_equilibrium to calculate the PfPR2-10 values for the range of
+# EIR values
+malEq_prev <- vapply(
+  malEq_EIR,
+  function(eir) {
+    eq <- malariaEquilibrium::human_equilibrium(
+      eir,
+      ft = 0,
+      p = eq_simparams,
+      age = 0:100
+    )
+    sum(eq$states[3:11, 'pos_M']) / sum(eq$states[3:11, 'prop'])
+  },
+  numeric(1)
+)
+
+# Establish a dataframe containing the matching EIR and PfPR2-10 values
+malEq_P2E <- cbind.data.frame(EIR = malEq_EIR, prev = malEq_prev)
```

Comment:
Print out the first 5-10 lines of `malEq_P2E` in the .Rmd so readers can visualize the output 😄

Reply:
Done - now prints out the first 7 rows so that the length matches the malsim example which has only 7 rows.